libs/lambda-edge-openid-auth/src/lib/utils/auth-token.ts

```diff
@@ -121,8 +121,13 @@ export function verifyToken(
         header: { kid },
     } = decodeToken(log, token)
 
+    if (!kid) {
+        log.warn({ kid }, 'Missing kid in token')
+        throw new Unauthorized('Missing kid')
+    }
+
     log.info('Searching for JWK from discovery document')
-    const pem = idpConfig.keyIdLookup[kid || '']
+    const pem = idpConfig.keystore.get(kid).toPEM()
     if (!pem) {
```

Comment:
If the key doesn't exist in the keystore would this cause an error when trying to do `.toPem()`.